ovs/extensions/support/agent.py

```diff
@@ -34,24 +33,128 @@
 from ovs.extensions.services.servicefactory import ServiceFactory
 
 
+class ConfigurationNotFoundError(RuntimeError):
+    pass
+
+
 class SupportAgent(object):
     """
     Represents the Support client
     """
-    _logger = Logger('extensions-support')
+    logger = Logger('extensions-support')
+
+    _MISSING = object()  # Used to check missing properties in function
+
+    DEFAULT_INTERVAL = 60
+    DEFAULT_ERROR_DELAY = 10  # Seconds to wait before rerunning loop on errors
+    DEFAULT_SUPPORT_ENABLED = False
+    DEFAULT_REMOTE_ACCESS_ENABLED = False
+
+    KEY_SUPPORT_AGENT = 'support_agent'
+    KEY_REMOTE_ACCESS = 'remote_access'
+    KEY_INTERVAL = 'interval'
+    FALLBACK_CONFIG = '/opt/OpenvStorage/config/support_agent.json'
+    LOCATION_CLUSTER_ID = '/ovs/framework/cluster_id'
+    LOCATION_INTERVAL = '/ovs/framework/support|{0}'.format(KEY_INTERVAL)
+    LOCATION_SUPPORT_AGENT = '/ovs/framework/support|{0}'.format(KEY_SUPPORT_AGENT)
+    LOCATION_REMOTE_ACCESS = '/ovs/framework/support|{0}'.format(KEY_REMOTE_ACCESS)
 
     def __init__(self):
         """
         Initializes the client
         """
-        self._cluster_id = Configuration.get('/ovs/framework/cluster_id').replace(r"'", r"'\''")
-        self._service_type = ServiceFactory.get_service_type()
-        self._storagerouter = System.get_my_storagerouter()
+        # Safe calls
+        self._node_id = System.get_my_machine_id().replace(r"'", r"'\''")
         self._package_manager = PackageFactory.get_manager()
         self._service_manager = ServiceFactory.get_manager()
-        self._client = SSHClient(endpoint=self._storagerouter)
+        self._service_type = ServiceFactory.get_service_type()
+
+        # Potential failing calls
+        try:
+            self._cluster_id = self.get_config_key(self.LOCATION_CLUSTER_ID, fallback=[Configuration.CONFIG_STORE_LOCATION, 'cluster_id'])
+        except ConfigurationNotFoundError:
```

Comment:
`ConfigurationNotFoundError` inherits from RuntimeError and its message is pretty much identical to 1 being raised in `self.get_config_key`